### PR TITLE
Fixes crossAxisSpacing overflow in RTL

### LINKED ISSUE #27151: GridView has unexpected overflow in RTL
`GridView` `crossAxisSpacing` Overflow RTL

When I'm using RTL layout, `GridView` and `crossAxisSpacing`, the children on right overflow the edge of the emulator.

### LTR

screenshot_1549008583

### RTL

screenshot_1549008589

```dart
new GridView.count(
            crossAxisCount: 2,
            padding: const EdgeInsets.all(12.0),
            mainAxisSpacing: 12.0,
            crossAxisSpacing: 12.0,
            children: new List<Widget>.generate(16, (index) {
              return new Container(color: Colors.red,child: new Text(""),);
            }),
          ),
```

### PR DESCRIPTION
**Pull request by p30arena**
Fixes https://github.com/flutter/flutter/issues/27151

### CODE REVIEW

**Comment by p30arena** [comment]
@goderbauer 
The new fix for the issue

**Review by goderbauer — COMMENTED** [review]
This looks like a good fix. Thanks!

Can you please add a test for this to make sure that we never break this again in the future?

**Comment by p30arena** [comment]
@goderbauer 
I added the tests.

File: packages/flutter/test/widgets/grid_view_test.dart [hunk 1]
```diff
@@ -536,4 +536,31 @@ void main() {
     expect(tester.getTopLeft(find.byKey(target)), const Offset(600.0, 0.0));
     expect(tester.getBottomRight(find.byKey(target)), const Offset(800.0, 200.0));
   });
+
+  testWidgets('GridView crossAxisSpacing', (WidgetTester tester) async {
```
Comment (goderbauer) [inline packages/flutter/test/widgets/grid_view_test.dart]: Can you please add the following comment after this line:

```suggestion
  testWidgets('GridView crossAxisSpacing', (WidgetTester tester) async {
    // Regression test for https://github.com/flutter/flutter/issues/27151.

**Review by goderbauer — COMMENTED** [review]
LGTM with one minor nit.

Thanks for fixing this. It's ready to land after you fix that one comment.

**Comment by p30arena** [comment]
@goderbauer
Done

**Comment by slightfoot** [comment]
@goderbauer I just came across this issue and resolved it before finding this issue/PR. I feel I have resolved it in a slightly clearer way. It might be worth checking it out. @p30arena sorry not trying to step on your toes. I copied your test case, so this can be merged/PRd or whatever. #28100

**Comment by p30arena** [comment]
@slightfoot 
Your solution is absolutely better.
BTW my toe hurts :smile:

**Comment by p30arena** [comment]
Closing this PR.
https://github.com/flutter/flutter/pull/28100

**Comment by p30arena** [comment]
@goderbauer 
Is this PR going to get merged any time soon?

**Comment by goderbauer** [comment]
Yeah, our tree is currently red (see flutter-build check above). As soon as it goes green, this can be merged.

**Pull request merged**